source/extensions/io_socket/user_space/io_handle_impl.cc

```diff
@@ -79,6 +79,10 @@ Api::IoCallUint64Result IoHandleImpl::close() {
       ENVOY_LOG(trace, "socket {} close after peer closed.", static_cast<void*>(this));
     }
   }
+  if (user_file_event_) {
+    // No event callback should be handled after close completes.
+    user_file_event_.reset();
+  }
```

Comment:
Please fix the PR title and description.  This fix is not to userspace IO handle, it is to the real io handle implementation.

Reply:
Sorry, I don't get your point: This fix is specific to userpspace IO handle. 

The OS io handle has this event clear during the close already. See https://github.com/envoyproxy/envoy/blob/main/source/common/network/io_socket_handle_impl.cc#L72-L75

Reply:
You're right.  Sorry.